src/Host/Client/Impl/Host/RHost.cs

```diff
@@ -375,19 +338,27 @@ public sealed partial class RHost : IDisposable, IRExpressionEvaluator {
                                 break;
 
                             case "?":
-                                await ShowDialog(message, allowEval, MessageButtons.YesNoCancel, CancellationTokenSource.CreateLinkedTokenSource(ct, _cancelAllCts.Token).Token);
+                                ShowDialog(message, MessageButtons.YesNoCancel, CancellationTokenSource.CreateLinkedTokenSource(ct, _cancelAllCts.Token).Token)
+                                    .SilenceException<MessageTransportException>()
+                                    .DoNotWait();
                                 break;
 
                             case "??":
-                                await ShowDialog(message, allowEval, MessageButtons.YesNo, CancellationTokenSource.CreateLinkedTokenSource(ct, _cancelAllCts.Token).Token);
+                                ShowDialog(message, MessageButtons.YesNo, CancellationTokenSource.CreateLinkedTokenSource(ct, _cancelAllCts.Token).Token)
```

Comment:
We have to, otherwise the loop won't process further incoming messages. Since the R inner loop won't move forward until it gets a response, it would seem okay?